### ISSUE TITLE
select with optgroups returns wrong value

### ISSUE THREAD

**Issue by philipraets** [Dogfalo/materialize#2159]
if I have a select with optgroups, the wrong value gets returned.
Everytime it encouters an optgroup, the results shifts up 1 position
codepen:
http://codepen.io/anon/pen/YyrqOE

**Comment by philipraets** [comment]
My guess is to change in forms.js this
```
      options.find('li').each(function (i) {
        var $curr_select = $select;
        $(this).click(function (e) {
          // Check if option element is disabled
          if (!$(this).hasClass('disabled') && !$(this).hasClass('optgroup')) {
            if (multiple) {
              $('input[type="checkbox"]', this).prop('checked', function(i, v) { return !v; });
              toggleEntryFromArray(valuesSelected, $(this).index(), $curr_select);
              $newSelect.trigger('focus');

            } else {
              options.find('li').removeClass('active');
              $(this).toggleClass('active');
              $curr_select.siblings('input.select-dropdown').val($(this).text());
            }

            activateOption(options, $(this));
            $curr_select.find('option').eq(i).prop('selected', true);
            // Trigger onchange() event
            $curr_select.trigger('change');
            if (typeof callback !== 'undefined') callback();
          }

          e.stopPropagation();
        });
      });
```
to this:
```
      options.find('li:not(.optgroup)').each(function (i) {
        var $curr_select = $select;
        $(this).click(function (e) {
          // Check if option element is disabled
          if (!$(this).hasClass('disabled')) {
            if (multiple) {
              $('input[type="checkbox"]', this).prop('checked', function(i, v) { return !v; });
              toggleEntryFromArray(valuesSelected, $(this).index(), $curr_select);
              $newSelect.trigger('focus');

            } else {
              options.find('li').removeClass('active');
              $(this).toggleClass('active');
              $curr_select.siblings('input.select-dropdown').val($(this).text());
            }

            activateOption(options, $(this));
            $curr_select.find('option').eq(i).prop('selected', true);
            // Trigger onchange() event
            $curr_select.trigger('change');
            if (typeof callback !== 'undefined') callback();
          }

          e.stopPropagation();
        });
      });
```

so the index count is correct

**Comment by acburst** [comment]
Fixed in 8ca298c. Thanks for pointing this out
